app/api/src/players/players.service.ts

```diff
@@ -42,4 +43,129 @@ export class PlayersService {
 		}
 		return player;
 	}
+
+	async findTop(topPlayersQuery: TopPlayersQueryDto) {
+		const { stat: hitsStat, gameType, minGames, minShots, limit } = topPlayersQuery;
+
+		const shotsStat = {
+			discDmgHitsTG: 'discShotsFiredTG',
+			discHitsTG: 'discShotsFiredTG',
+			discMATG: 'discShotsFiredTG',
+			laserHitsTG: 'laserShotsFiredTG',
+			laserMATG: 'laserShotsFiredTG',
+			cgHitsTG: 'cgShotsFiredTG',
+			shockHitsTG: 'shockShotsFiredTG'
+		}[hitsStat]
+
+		// Possibly make this a query param at some point?
+		const excludeDiscJumps = shotsStat === 'discShotsFiredTG';
+
+		const hitsValue = '(game.stats->:hitsStat->>0)::integer';
+		const shotsValue = '(game.stats->:shotsStat->>0)::integer';
+		const discJumpsValue = "(game.stats->'discJumpTG'->>0)::integer";
+		const killerDiscJumpsValue = "(game.stats->'killerDiscJumpTG'->>0)::integer";
```

Comment:
It seems fast enough, but some indexes might make it faster.

e.g. on `gametype`, and whatever indexing is possible on the `jsonb` content.

Reply:
Agreed. I haven't dug too deeply on performance tuning yet as it did seem fast enough. Reasonings for `jsonb` col type is that the shape of the available stats were changing pretty frequently and this allowed for flexibility. The `jsonb` also allows for 3 different types of indexes - gin, btree and hash where as the plain `json` col type does not.